chapter-05/layered/money/decimal.go

```diff
@@ -0,0 +1,78 @@
+package money
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// number is a structure that can hold a number with a fixed precision.
+// example: 1.52 = 1 + 52 * 10^(-2) will be stored as {1, 52, 2}
+type number struct {
+	// the integer part of the number
+	integerPart int
+	// the decimal part of the number
+	decimalPart int
+	// precision of the decimal part, as the exponent of a power of 10
+	precision int
+}
+
+const (
+	// errors that we might face when parsing numbers
+	errInvalidInteger = moneyError("unable to convert integer part")
+	errInvalidDecimal = moneyError("unable to convert decimal part")
+)
+
+// parseNumber converts a string into its number representation.
+// It assumes there is up to one decimal separator, and that the separator is '.' (full stop character).
+func parseNumber(value string) (number, error) {
+	intPart, decPart, found := strings.Cut(value, ".")
+
+	i, err := strconv.Atoi(intPart)
```

Comment:
we need to test `-1`, `.25`, `+5`, `4.`, `5.0`

Reply:
- [ ] @doniacld add tests cases